src/main/duke/src/Task.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Handles Task
+ * Super class of Deadline, Event, and ToDo
+ */
+public class Task {
+    protected  String description;
+    protected boolean isDone;
+
+    /**
+     * Constructor of Task
+     * @param description
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Determines whether the task is done or not
+     * @return String about whether the task is done or not
+     */
+    public String getStatusIcon() {
+        return (isDone ? "[X]" : "[ ]"); // mark done task with X
+    }
+
+    /**
+     *
+     * @return String of status icon and description of task
+     */
+    @Override
+    public String toString() {
+        return getStatusIcon() + " " + this.description;
+    }
+
+    /**
+     * Changes protected boolean isDone to true
+     */
+    public void markAsDone() {
+        this.isDone = true;
+        System.out.println("Nice! I've marked this task as done:");
```

Comment:
Since the `Ui` class is supposed to handle the user input and interface, consider making use of `Ui` to do the printing instead!